### PR TITLE
evm: add external quoteDeliveryPrice()

### PR DESCRIPTION
**Pull request by panoel**

### CODE REVIEW

File: evm/src/Router.sol [hunk 1]
```diff
@@ -563,6 +573,25 @@ contract Router is IRouterAdmin, IRouterIntegrator, IRouterTransceiver, MessageS
         _;
     }
 
+    /// @notice Retrieves the quote for message delivery.
+    /// @dev This version does not need to be called by the integrator.
+    /// @dev This sums up all the individual recvTransceiver's quoteDeliveryPrice calls.
+    /// @param integrator The address of the integrator.
```
[the remark this answers is not in the document]
Comment (panoel) [reply evm/src/Router.sol:579]: Fixed

File: evm/src/Router.sol [hunk 2]
```diff
@@ -563,6 +573,25 @@ contract Router is IRouterAdmin, IRouterIntegrator, IRouterTransceiver, MessageS
         _;
     }
 
+    /// @notice Retrieves the quote for message delivery.
```
Comment (evan-gray) [inline evm/src/Router.sol:576]: Why is this comment duplicated instead of inherited?

**Review by bruce-riley — DISMISSED** [review]

File: evm/src/interfaces/IRouterIntegrator.sol [hunk 3]
```diff
@@ -67,4 +67,19 @@ interface IRouterIntegrator is IMessageSequence {
...
         external
         returns (uint128, uint128, bool);
+
+    /// @notice Retrieves the quote for message delivery.
+    /// @dev This version does not need to be called by the integrator.
+    /// @dev This sums up all the individual sendTransceiver's quoteDeliveryPrice calls.
+    /// @param integrator The address of the integrator.
+    /// @param dstChain The Wormhole chain ID of the recipient.
+    /// @return uint256 The total cost of delivering a message to the recipient chain in this chain's native token.
+    function quoteDeliveryPrice(address integrator, uint16 dstChain) external returns (uint256);
```
Comment (bruce-riley) [inline evm/src/interfaces/IRouterIntegrator.sol:77]: I think these should be marked view like they are in Router.sol.

Comment (panoel) [reply evm/src/interfaces/IRouterIntegrator.sol:78]: Changed

**Review by bruce-riley — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**